### PR TITLE
feat: add delete sub-command for backup-location

### LINKED ISSUE #2168: No cli option to delete a backup location (Velero 1.1.0)
**What steps did you take and what happened:**

- create backup location as follows:

```
velero backup-location create source-cluster --provider azure --bucket mycontainer --access-mode=ReadOnly
```

- locations are created:

```
[aks-velero]$ velero get backup-locations
NAME             PROVIDER   BUCKET/PREFIX     ACCESS MODE
default          azure      hercontainer   ReadWrite
source-cluster   azure      mycontainer   ReadOnly
```

- Try to delete the backup location

```
[ aks-velero]$ velero backup-location delete source-cluster
Work with backup storage locations

Usage:
  velero backup-location [command]
....
```

**What did you expect to happen:**

- backup location metadata/configuration is deleted.

**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

* `kubectl logs deployment/velero -n velero`

```
time="2019-12-27T05:33:55Z" level=info msg="setting log-level to INFO" logSource="pkg/cmd/server/server.go:168"
time="2019-12-27T05:33:55Z" level=info msg="Starting Velero server v1.1.0 (a357f21aec6b39a8244dd23e469cc4519f1fe608)" logSource="pkg/cmd/server/server.go:170"
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/pod
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/pv
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/service-account
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=VolumeSnapshotter logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/aws
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=VolumeSnapshotter logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/azure
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=VolumeSnapshotter logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/gcp
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=ObjectStore logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/aws
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=ObjectStore logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/azure
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=ObjectStore logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/gcp
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/add-pv-from-pvc
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/add-pvc-from-pod
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/change-storage-class
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/job
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/pod
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/restic
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/service
time="2019-12-27T05:33:55Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/service-account
time="2019-12-27T05:33:55Z" level=info msg="Checking existence of namespace" logSource="pkg/cmd/server/server.go:336" namespace=velero
time="2019-12-27T05:33:55Z" level=info msg="Namespace exists" logSource="pkg/cmd/server/server.go:342" namespace=velero
time="2019-12-27T05:33:58Z" level=info msg="Checking existence of Velero custom resource definitions" logSource="pkg/cmd/server/server.go:371"
time="2019-12-27T05:33:58Z" level=info msg="All Velero custom resource definitions exist" logSource="pkg/cmd/server/server.go:405"
time="2019-12-27T05:33:58Z" level=info msg="Checking that all backup storage locations are valid" logSource="pkg/cmd/server/server.go:412"
An error occurred: some backup storage locations are invalid: error getting backup store for location "source-cluster": rpc error: code = Unknown desc = unable to get all required config values: the following keys do not have values: resourceGroup, storageAccount
```

**Anything else you would like to add:**

- I'd like to update the configuration for the backup location but it's unclear how to do so without the ability to delete and re-create  it ...

**Environment:**

- Velero version (use `velero version`): 

```
[$ velero version
Client:
        Version: v1.1.0
        Git commit: a357f21aec6b39a8244dd23e469cc4519f1fe608

```

- Velero features (use `velero client config get features`): 

```
[ aks-velero]$ velero client config get features
features: <NOT SET>
```

- Kubernetes version (use `kubectl version`):

```
kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-06T01:44:30Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.8", GitCommit:"1da9875156ba0ad48e7d09a5d00e41489507f592", GitTreeState:"clean", BuildDate:"2019-11-14T05:19:20Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```

- Kubernetes installer & version:

Azure AKS

- Cloud provider or hardware configuration:

Azure

- OS (e.g. from `/etc/os-release`):

CentOS 7

### PR DESCRIPTION
**Pull request by jenting**
fixes #2168 

1. Add delete sub-command for BSL, list available sub-commands
```
velero backup-location delete <NAME>
```
or
```
velero delete backup-location <NAME>
velero delete backup-locations <NAME>
```

2. Fix use label selector to get BSL doesn't work.
3. Makes the --help message format alignments.

### CODE REVIEW

File: pkg/cmd/cli/backuplocation/delete.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	}
+
+	var (
+		backupLocations []*velerov1api.BackupStorageLocation
+		errs            []error
+	)
+	switch {
+	case len(o.Names) > 0:
+		for _, name := range o.Names {
+			backupLocation, err := o.Client.VeleroV1().BackupStorageLocations(o.Namespace).Get(context.TODO(), name, metav1.GetOptions{})
```
Comment (carlisia) [inline pkg/cmd/cli/backuplocation/delete.go:81]: I'd like to ask you to change the API calls to use the kubebuilder/runtimecontroller API. Here's an example:

https://github.com/vmware-tanzu/velero/blob/main/pkg/cmd/cli/backuplocation/get.go#L48-L61

**Review by carlisia — CHANGES_REQUESTED** [review]
This is a great add, thank you! I tested it and it works, but I do have a request to use the new api calls.

**Comment by jenting** [comment]
This is ready to be reviewed.

**Review by carlisia — APPROVED** [review]
I tested this and it works as intended. Thank you for this contribution!

**Review by nrb — APPROVED** [review]

**Pull request merged**